packages/taiko-client/driver/chain_syncer/blob/syncer.go

```diff
@@ -475,12 +475,9 @@ func (s *Syncer) checkReorg(ctx context.Context, blockID *big.Int) (*rpc.ReorgCh
 		return nil, fmt.Errorf("failed to check if last verified block in L2 EE has been reorged: %w", err)
 	}
 
-	if reorgCheckResult == nil {
-		// 2. Parent block
-		reorgCheckResult, err = s.rpc.CheckL1Reorg(
-			ctx,
-			new(big.Int).Sub(blockID, common.Big1),
-		)
+	// 2. If the verified block check is passed, we check the parent block.
+	if reorgCheckResult == nil || !reorgCheckResult.IsReorged {
```

Comment:
Is the `reorgCheckResult == nil ` case imposiable?

Reply:
Yes, just for safety, as suggested in https://github.com/taikoxyz/taiko-mono/pull/19166#discussion_r2015343862